sdk/platform/http_client/curl/src/az_curl.c

```diff
@@ -478,18 +476,23 @@ _az_http_client_curl_setup_url(CURL* p_curl, _az_http_request const* p_request)
   AZ_PRECONDITION_NOT_NULL(p_curl);
   AZ_PRECONDITION_NOT_NULL(p_request);
 
+  az_span request_url = { 0 };
+  // get request_url. It will have the size of what is has written in it only
+  AZ_RETURN_IF_FAILED(az_http_request_get_url(p_request, &request_url));
+  int32_t request_url_size = az_span_size(request_url);
+
   az_span writable_buffer;
   {
     // Add 1 for 0-terminated str
-    int32_t const url_final_size = p_request->_internal.url_length + 1;
+    int32_t const url_final_size = request_url_size + 1;
 
     // allocate buffer to add \0
     AZ_RETURN_IF_FAILED(_az_span_malloc(url_final_size, &writable_buffer));
   }
 
   // write url in buffer (will add \0 at the end)
-  az_result result = _az_http_client_curl_append_url(
-      writable_buffer, az_span_slice(p_request->_internal.url, 0, p_request->_internal.url_length));
+  // request_url is already with the size of what it has written only
```

Comment:
Can you clarify what are you trying to say here?

Reply:
inside an az_http_request we have one az_span for url and one int32_t for the `url_length`.
People might think that, after getting the url from request with `az_http_request_get_url()` they still need to slice that az_span with `url_length`. This comment say that url is already that slice because `az_http_request_get_url()` take cares of returning just that.

Reply:
```suggestion
  // request_url is already the right size containing only what has been written into it
```